src/couch_stats/test/eunit/csrt_logger_tests.erl

```diff
@@ -0,0 +1,368 @@
+        "csrt_logger.matchers_threshold", "rows_read", integer_to_list(?THRESHOLD_ROWS_READ), false
+    ),
+    ok = config:set(
+        "csrt_logger.matchers_threshold",
+        "worker_changes_processed",
+        integer_to_list(?THRESHOLD_CHANGES),
+        false
+    ),
+    ok = config:set("csrt_logger.dbnames_io", "foo", integer_to_list(?THRESHOLD_DBNAME_IO), false),
+    ok = config:set("csrt_logger.dbnames_io", "bar", integer_to_list(?THRESHOLD_DBNAME_IO), false),
+    ok = config:set(
+        "csrt_logger.dbnames_io", "foo/bar", integer_to_list(?THRESHOLD_DBNAME_IO), false
+    ),
+    csrt_logger:reload_matchers(),
+    #{ctx => Ctx, dbname => DbName, rctx => Rctx, rctxs => rctxs()}.
+
+teardown(#{ctx := Ctx, dbname := DbName}) ->
+    ok = fabric:delete_db(DbName, [?ADMIN_CTX]),
+    ok = meck:unload(ioq),
+    test_util:stop_couch(Ctx).
+
+rctx_gen() ->
+    rctx_gen(#{}).
+
+rctx_gen(Opts0) ->
+    DbnameGen = one_of([<<"foo">>, <<"bar">>, ?tempdb]),
+    TypeGen = one_of([?RCTX_RPC, ?RCTX_COORDINATOR]),
+    R = fun() -> rand:uniform(?RCTX_RANGE) end,
+    R10 = fun() -> 3 + rand:uniform(round(?RCTX_RANGE / 10)) end,
+    Occasional = one_of([0, 0, 0, 0, 0, R]),
+    Nonce = one_of(["9c54fa9283", "foobar7799", lists:duplicate(10, fun nonce/0)]),
+    Base = #{
+        dbname => DbnameGen,
+        db_open => R10,
+        docs_read => R,
+        docs_written => Occasional,
+        get_kp_node => R10,
+        get_kv_node => R,
+        nonce => Nonce,
+        pid_ref => {self(), make_ref()},
+        ioq_calls => R,
+        rows_read => R,
+        type => TypeGen,
+        %% Hack because we need to modify both fields
+        '_do_changes' => true
+    },
+    Opts = maps:merge(Base, Opts0),
+    csrt_util:map_to_rctx(
+        maps:fold(
+            fun
+                %% Hack for changes because we need to modify both changes_processed
+                %% and changes_returned but the latter must be <= the former
+                ('_do_changes', V, Acc) ->
+                    case V of
+                        true ->
+                            Processed = R(),
+                            Returned = (one_of([0, 0, 1, Processed, rand:uniform(Processed)]))(),
+                            maps:put(
+                                changes_processed,
+                                Processed,
+                                maps:put(changes_returned, Returned, Acc)
+                            );
+                        _ ->
+                            Acc
+                    end;
+                (K, F, Acc) when is_function(F) ->
+                    maps:put(K, F(), Acc);
+                (K, V, Acc) ->
+                    maps:put(K, V, Acc)
+            end,
+            #{},
+            Opts
+        )
+    ).
+
+rctxs() ->
+    [rctx_gen() || _ <- lists:seq(1, ?RCTX_COUNT)].
+
+t_do_report(#{rctx := Rctx}) ->
+    JRctx = csrt_util:to_json(Rctx),
+    ReportName = "foo",
+    ok = meck:new(couch_log),
```

Comment:
Can you move all `meck:new` calls to `setup` (see [here](https://pages.github.ibm.com/cloudant/dbcore/tutorial/pr.review.checklist.html#meck))? There are known issues with `meck` which leave mocked versions of the modules loaded. You can keep `meck:expect` here and also can use `meck:reset(couch_log)` to reset the history in each test.